packages/identity-service/src/authMiddleware.js

```diff
@@ -67,22 +122,39 @@ async function authMiddleware(req, res, next) {
         'handle'
       ]
     })
-    if (!user)
-      throw new Error(
-        `[Error]: no user found for wallet address ${walletAddress}`
-      )
-
-    // This overwrites any provisionally set handles (b/c blockchainUserId is never set with those),
-    // ensuring that the user.handle always represents the latest state on chain
-    if (!user.blockchainUserId || !user.handle) {
-      const discprovUser = await queryDiscprovForUserId(walletAddress, handle)
-      user = await user.update({
-        blockchainUserId: discprovUser.user_id,
-        handle: discprovUser.handle
-      })
+    // We are managing someone else
+    if (actingUserId && actingUserId !== user.blockchainUserId) {
+      try {
+        const actingUser = await getManagedUser(walletAddress, actingUserId, {
+          'Encoded-Data-Message': encodedDataMessage,
+          'Encoded-Data-Signature': signature
+        })
+        req.user = actingUser
+      } catch (err) {
+        const errorResponse = errorResponseForbidden(
+          `[Error]: The wallet address is not associated with a user id ${err}`
```

Comment:
nit: This message feels a little confusing in the context of the user not being able to act on behalf of the given user id

Reply:
doh!